### PR TITLE
Cutflow improvements

### PR DESCRIPTION
**Pull request by ireed3597**

### CODE REVIEW

**Comment by carriganm95** [comment]
Can you just post the printout of what this does? Or send me a quick script I can run to use these changes

**Comment by ireed3597** [comment]
Screenshot from 2024-01-19 15-37-03
Screenshot from 2024-01-19 15-37-23
Here you go.

File: Run3Detector/analysis/utilities/milliqanProcessor.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def makeBranches(self, events):
         #self.plotter.events = events
         self.mqSchedule.setEvents(events)
         for branch in self.mqSchedule.schedule:
+            print(branch.__name__)
```
Comment (carriganm95) [inline Run3Detector/analysis/utilities/milliqanProcessor.py:32]: can we remove this?

File: Run3Detector/analysis/utilities/milliqanCuts.py [hunk 2]
```diff
@@ -15,6 +39,17 @@ def combineCuts(self, name, cuts):
             else:
                 self.events[name] = self.events[cut]
 
+    # Dummy cut for use while construcing the cutflow mechanics
+    def neverCut(self, cutName=None, cut=False):
+        if cut: self.events = self.events
+        self.cutflowCounter()
+
+    # Dummy cut for use while construcing the cutflow mechanics
+    def idCut(self, cutName=None, cut=False):
```
Comment (carriganm95) [inline Run3Detector/analysis/utilities/milliqanCuts.py:48]: if this is no longer needed delete

File: Run3Detector/analysis/utilities/milliqanCuts.py [hunk 3]
```diff
@@ -6,6 +6,30 @@ class milliqanCuts():
 
     def __init__(self):
         self.events = []
+        self.cutflow = []
+        self.counter = 0
+
+    def cutflowCounter(self):
+        # Increments events passing each stage of the cutflow
+        # Creates each stage during the first pass
+        try:
```
Comment (carriganm95) [inline Run3Detector/analysis/utilities/milliqanCuts.py:15]: could this just be an if else?

`if self.counter < len(self.cutflow): 
    ...
else:
    self.cutflow.append`

File: Run3Detector/analysis/utilities/milliqanCuts.py [hunk 4]
```diff
@@ -6,6 +6,30 @@ class milliqanCuts():
...
+        # Builds the array without knowledge of the number of cuts
+        # TODO: See if there is a more efficent way to implement this
+        except:
+            self.cutflow.append(len(self.events))
+        self.counter+=1
+
+    def getCutflowCounts(self):
+        # Prints the value after each batch of events
+        # TODO: Only print at the very end
+        print(self.cutflow)
```
Comment (carriganm95) [inline Run3Detector/analysis/utilities/milliqanCuts.py:26]: remove this print before the actual cutflow printing?

File: Run3Detector/analysis/utilities/milliqanCuts.py [hunk 5]
```diff
@@ -6,6 +6,30 @@ class milliqanCuts():
...
+        print("Cutflow Table")
```
Comment (carriganm95) [inline Run3Detector/analysis/utilities/milliqanCuts.py:27]: Just to make things pretty
```suggestion
        print("------------------Cutflow Table--------------------")
        print ("{:<15} {:<10}".format('Cut', 'N Passing Events'))
        for i in range((len(self.cutflow)):
            print("{:<15}{:<10}".format(i, self.cutflow[i])
        print("----------------------------------------------------")
```

**Review by carriganm95 — COMMENTED** [review]
Looks good just some small clean up

File: Run3Detector/analysis/utilities/milliqanProcessor.py (hunk 1 above)
Comment (ireed3597) [reply Run3Detector/analysis/utilities/milliqanProcessor.py:32]: yep

File: Run3Detector/analysis/utilities/milliqanCuts.py (hunk 2 above)
Comment (ireed3597) [reply Run3Detector/analysis/utilities/milliqanCuts.py:48]: Removed, but I'd like to keep the ```neverCut``` for now, just as an extra debugging tool

File: Run3Detector/analysis/utilities/milliqanCuts.py (hunk 4 above)
Comment (ireed3597) [reply Run3Detector/analysis/utilities/milliqanCuts.py:26]: Done

**Comment by ireed3597** [comment]
image

Looks much nicer now

**Comment by carriganm95** [comment]
ok looks good, im going to add an issue to have a global on/off for the cutflow but that doesn't need to hold up this PR

**Review by carriganm95 — APPROVED** [review]

**Pull request merged**